src/renderers/dom/shared/ReactDOMComponent.js

```diff
@@ -687,6 +698,19 @@ ReactDOMComponent.Mixin = {
   },
 
   /**
+   * Setup this component to trap non-bubbling events locally
+   *
+   * @private
+   * @param {ReactReconcileTransaction|ReactServerRenderingTransaction} transaction
+   */
+  _setupTrapBubbledEventsLocal: function(transaction) {
+    this._wrapperState = {
+      listeners: null,
+    };
+    transaction.getReactMountReady().enqueue(trapBubbledEventsLocal, this);
```

Comment:
The `mountWrapper` calls in input, select, and textarea should include the `listeners` property in their `_wrapperState` to avoid changing the hidden class when the assignment is made. This would also be better just as a free function – we're trying to get away from OO in the core anyway.